node_modules/oae-admin/downloadlog/js/downloadlog.js

```diff
@@ -0,0 +1,227 @@
+/*!
+ * Copyright 2018 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['jquery', 'oae.core'], function($, oae) {
+
+    return function(uid, showSettings) {
+
+        // The widget container
+        var $rootel = $('#' + uid);
+
+        // Variable that keeps track of the current tenant context
+        var currentContext = null;
+
+        /**
+         * Bind all the action listeners needed for the user to interact with the "download" panel in
+         */
+        var bindListeners = function() {
+            // When the user valide the form, download file
+            $rootel.on('click', '#submit-download-btn', function(event) {
+                // Avoid the reload
+                event.preventDefault();
+                downloadLog(function(err) {
+                    setTimeout(function() {
+                        closeModal();
+                    }, 100);
+                    // Avoid default form submit behavior
+                    return false;
+                });
+            });
+
+            // When the user click on cancel
+            $rootel.on('click', '#cancel-download-btn', function() {
+                // Avoid default form submit behavior
+                return false;
+            });
+
+            $rootel.on('keydown', '#download-log-input', function(e) {
+                // Allow: backspace, delete, tab, escape, enter, left arrow and right arrow
+                if ($.inArray(e.keyCode, [46, 8, 9, 27, 13, 110, 37, 39]) !== -1 ||
+                    // Allow: Ctrl/cmd+A
+                    (e.keyCode == 65 && (e.ctrlKey === true || e.metaKey === true)) ||
+                    // Allow: Ctrl/cmd+C
+                    (e.keyCode == 67 && (e.ctrlKey === true || e.metaKey === true)) ||
+                    // Allow: Ctrl/cmd+X
+                    (e.keyCode == 88 && (e.ctrlKey === true || e.metaKey === true)) ||
+                    // Allow: numpad 
+                    (e.keyCode >= 96 && e.keyCode <= 120 && e.shiftKey) ||
+                    // Allow: 0-9 
+                    (e.keyCode >= 48 && e.keyCode <= 57 && e.shiftKey)) {
+                        // let it happen, don't do anything
+                        return;
+                }
+                // Ensure that it is a number and stop the keypress
+                if ((!e.shiftKey) || (e.keyCode < 48 || e.keyCode > 57)) {
```

Comment:
Also, make sure this is being protected in the backend as well. I'm not sure it is, can't remember